### PR TITLE
Added `OperatorDeclaration`

### PR DESCRIPTION
**Pull request by oxisto**
This PR adds support for an operator declaration. They are not resolved yet. Resolution of overloaded operators will be tackled in #1606

### CODE REVIEW

File: cpg-language-cxx/src/main/kotlin/de/fraunhofer/aisec/cpg/frontends/cxx/DeclaratorHandler.kt [hunk 1]
```diff
@@ -168,9 +167,14 @@ class DeclaratorHandler(lang: CXXLanguageFrontend) :
 
         val func =
             when {
+                // Check if it's an operator. In this case, the name begins with operator
+                name.localName.startsWith("operator") -> {
```
Comment (konradweiss) [inline cpg-language-cxx/src/main/kotlin/de/fraunhofer/aisec/cpg/frontends/cxx/DeclaratorHandler.kt:171]: Some change is needed here: the comparison of operator with language trait operator names

File: cpg-language-cxx/src/main/kotlin/de/fraunhofer/aisec/cpg/frontends/cxx/DeclaratorHandler.kt [hunk 2]
```diff
@@ -219,7 +223,8 @@ class DeclaratorHandler(lang: CXXLanguageFrontend) :
 
         /*
          * As always, there are some special cases to consider and one of those are C++ operators.
-         * They are regarded as functions and eclipse CDT for some reason introduces a whitespace in the function name, which will complicate things later on
+         * They are regarded as functions and eclipse CDT for some reason introduces a whitespace
+         * in the function name, which will complicate things later on
          */
         if (name.startsWith("operator")) {
```
Comment (konradweiss) [inline cpg-language-cxx/src/main/kotlin/de/fraunhofer/aisec/cpg/frontends/cxx/DeclaratorHandler.kt:229]: Replace only if after replacement it is contained in the language operators.

**Review by konradweiss — CHANGES_REQUESTED** [review]

File: cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/frontends/LanguageTraits.kt [hunk 3]
```diff
@@ -230,3 +232,13 @@ interface HasFunctionalCasts : LanguageTrait
...
 interface HasFunctionOverloading : LanguageTrait
+
+/** A language trait that specifies that this language allows overloading of operators. */
+interface HasOperatorOverloading : LanguageTrait {
+
+    /**
+     * The name of the operators. The key is the [HasOperatorCode.operatorCode] and the value is the
+     * name of the operator.
+     */
+    val operatorNames: Map<String, Symbol>
```
Comment (oxisto) [inline cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/frontends/LanguageTraits.kt:243]: After initial confusion: Should we rename this to `overloadedOperatorNames`? @maximiliankaul

Comment (maximiliankaul) [reply cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/frontends/LanguageTraits.kt:243]: Yes, please :)

**Review by maximiliankaul — APPROVED** [review]

**Pull request merged**